src/components/NewClusterModal.js

```diff
@@ -127,6 +130,13 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
     }
   }
 
+  deleteCluster() {
+    const { currentCluster, onSuccess } = this.props
+    const { googleProject, clusterName } = currentCluster
+
+    onSuccess(Ajax().Jupyter.cluster(googleProject, clusterName).delete())
```

Comment:
Calling `onSuccess` here results in it getting called twice in the 'create' case, since `createCluster` calls it already. Also, nothing is being returned here, so the `Promise.all` isn't going to do the same thing as before. If we just returned the promise here, then we could call `onSuccess` in the click handler for the delete button, and everything should behave as expected.